app/src/docs/components/PageTurner/index.jsx

```diff
@@ -0,0 +1,91 @@
+// @flow
+
+import React from 'react'
+import { Link, withRouter, type Location } from 'react-router-dom'
+import cx from 'classnames'
+import { formatPath } from '$shared/utils/url'
+import SvgIcon from '$shared/components/SvgIcon'
+
+import type { NavigationLink } from '$docs/flowtype/navigation-types'
+
+import styles from './pageTurner.pcss'
+
+type Props = {
+    navigationItems: NavigationLink,
+    location: Location,
+}
+
+class PageTurner extends React.Component<Props> {
+    generateNavButtons() {
+        const { navigationItems } = this.props
+        const { pathname } = this.props.location
+
+        let pathMatch = false
+        let firstMatch = false
+
+        const navigationPaths = Object.entries(navigationItems).reduce((acc, [key, val], index) => {
+            let result = {
+                ...acc,
+            }
+
+            if (val === pathname) {
+                pathMatch = true
+            }
+
+            if (pathMatch && index === 0) {
+                firstMatch = true
+                return result
+            }
+
+            switch (Object.keys(result).length) {
+                case 0:
+                    if (val !== pathname) {
+                        result[key] = val
+                    }
+                    return result
+                case 1:
+                    if (!firstMatch) {
+                        if (val !== pathname && !pathMatch) {
+                            result = {}
+                            result[key] = val
+                        } else if (val !== pathname && pathMatch) {
+                            result[key] = val
+                        }
+                    }
+                    return result
+                default:
+                    return result
+            }
+        }, {})
+
+        return (
+            Object.entries(navigationPaths).map(([linkTitle, linkPath], index) => (
+                <li
+                    key={`item-${String(linkTitle)}`}
+                    className={cx(styles.navButton, {
+                        [styles.backward]: index === 0 && !firstMatch,
+                        [styles.forward]: index === 1 || firstMatch,
+                    })}
+                >
+                    <Link to={formatPath(String(linkPath))}>
+                        {(index === 0 && !firstMatch) ? 'Back to ' : ''}{linkTitle}
+                        <SvgIcon
+                            name="back"
+                            className={styles.arrow}
+                        />
+                    </Link>
+                </li>
+            ))
+        )
+    }
+
+    render() {
+        return (
+            <ul className={styles.pageTurnerContainer}>
+                {this.generateNavButtons()}
+            </ul>
+        )
+    }
+}
+
+export default withRouter(PageTurner)
```

Comment:
Usually I don't like using withRouter because of https://github.com/ReactTraining/react-router/pull/5552#issuecomment-331502281. But if it's hard to pass location down from parent(s) maybe this is ok 🙂

Reply:
@tumppi I had a go at this but my problem is that Storybook has trouble with the full demonstration of the component without a working router. For some reason storyRouter isn't getting the job done there! 
Any ideas there or do you think I should I just revert back to using withRouter()?

Reply:
Ah, don't know about storyRouter. Maybe you should revert back to withRouter 👍

Reply:
Sounds good, will keep an eye on withRouter() usage in the future anyway, good to know! 👍